src/NuGet.Core/NuGet.Protocol/Resources/PackageSearchResourceV3.cs

```diff
@@ -61,7 +61,7 @@ public override async Task<IEnumerable<IPackageSearchMetadata>> SearchAsync(stri
                     filter,
                     skip,
                     take,
-                    Common.NullLogger.Instance,
+                    log,
```

Comment:
Can you elaborate here?

Reply:
I was passing the Console as the logger to SearchAsync(), but instead of passing this logger on to the next Search method, it passed a null logger, so anytime the logger was called to do anything, it was useless, and there was no point in having the logger parameter in the first place. This way, SearchAsync will pass on the logger as the caller intended, and if the caller wants a null logger to be passed on, it can pass that directly as a parameter.